templates/addons/aws/modules-v5/cloudtrail/s3-bucket-cloudtrail/main.tf

```diff
@@ -0,0 +1,162 @@
+data "aws_caller_identity" "current" {}
+data "aws_partition" "current" {}
+data "aws_region" "current" {}
+
+locals {
+  bucket_name = var.s3_bucket_name
+  trail_arns  = [for name in var.trail_names : "arn:${data.aws_partition.current.partition}:cloudtrail:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:trail/${name}"]
+}
+
+# trivy:ignore:AVD-AWS-0089 Access logging not required for CloudTrail log bucket
+resource "aws_s3_bucket" "cloudtrail_logs" {
```

Comment:
We are duplicating S3 code in a few modules. Should we adjust the current S3 module, or create a new S3 module that we can use for these purposes? What do you think?